src/Elastic.Apm/Agent.cs

```diff
@@ -96,12 +96,13 @@ public static class Agent
 	{
 		private static readonly Lazy<ApmAgent> LazyApmAgent = new Lazy<ApmAgent>(() => new ApmAgent(_components));
 		private static AgentComponents _components;
+		private static volatile bool _isConfigured;
 
 		public static IConfigurationReader Config => Instance.ConfigurationReader;
 
 		internal static ApmAgent Instance => LazyApmAgent.Value;
 
-		public static bool IsConfigured { get; private set; }
+		public static bool IsConfigured => _isConfigured;
```

Comment:
Adding `volatile` backing field achieves absolutely nothing as described in the PR.  From [here](http://www.albahari.com/threading/part4.aspx#_The_volatile_keyword):
> As it happens, Intel’s X86 and X64 processors always apply acquire-fences to reads and release-fences to writes — whether or not you use the **volatile** keyword — so this keyword has no effect on the _hardware_ if you’re using these processors. However, **volatile** _does_ have an effect on optimizations performed by the compiler and the CLR — as well as on 64-bit AMD and (to a greater extent) Itanium processors. This means that you cannot be more relaxed by virtue of your clients running a particular type of CPU.
> 
> (And even if you _do_ use volatile, you should still maintain a healthy sense of anxiety, as we’ll see shortly!).

I see no point in adding this backing field for the above reasons.  It will not solve anything to do with thread-safety, and will only add complexity and overhead to the code.